### DISCUSSION TITLE
Bespoke Recipes and Where to Put Them

### DISCUSSION THREAD

**Discussion by thefennecdeer** [museumsvictoria/nodel#304] [General]
Hey all!
I imagine a lot of us have racked up quite a few recipes that are extremely specific or otherwise one-time use, and while I wouldn't want my questionable ones clogging up the official recipe repo (especially if it's getting pulled to each new Nodel install), I still think there'd be a benefit to sharing them, even if they're minor edits of existing recipes.

Something like recipes repo we already have, but seperated and more in-depth - possibly like those [Awesome lists](https://github.com/dictcp/awesome-git) in the main readme, with each folder having another readme.md for each category or topic.

It'd be also good for seeing how creative people are getting with what Nodel can do.
_Also I need more organisations to start using Nodel to control Meta Quest 2s to justify how much time I've spent working on that one python file._

So, if this were to exist, would you guys personally use it/look through it?

**Comment by dargs** [comment]
Not sure the answer to the actual question, but I'm curious what features on the Meta Quest 2 you are controlling/monitoring with Nodel !?

**Reply by thefennecdeer** [reply]
We've got Quest 2s connected to computers, so we can use them as if they were just the headset part (with the PC driving the visuals). Unfortunately, there's a whole android operating system there that really wants to act standalone.

So, I've taken advantage of the Android Debugging Bridge tools to tell the Quests that we want them to receive visuals from the PC - called Quest Link.

You can have a look at the code [here](https://github.com/thefennecdeer/Nodes/blob/main/EX%20VR%20Node.py), but the gist is that we're telling the headset to launch the Quest Link App, then repeatedly querying the current 'focused' app. There's also some nice extra things like disabling the home-shell to get rid of notifications, and toggling the proximity sensor during the first launch (when the headset isn't being worn it doesn't behave predictably, so we tell the headset that someones wearing it for the duration of the setup). I've got all this hooked up to signals so we get a nice overview on a dashboard.
![image](https://github.com/museumsvictoria/nodel/assets/28147677/4426ade9-09f1-4acf-823b-cbd189210ea3)
